src/stability_sdk/client.py

```diff
@@ -333,17 +455,51 @@ def upscale(
             parameters=[generation.StepParameter(**step_parameters)],
         )
 
-        prompts = [image_to_prompt(init_image)]
+        request_id = str(uuid.uuid4())
 
-        if prompt:
-            if isinstance(prompt, str):
-                prompt = generation.Prompt(text=prompt)
-            elif not isinstance(prompt, generation.Prompt):
-                raise ValueError("prompt must be a string or Prompt object")
-            prompts.append(prompt)
+        rq = generation.Request(
+            engine_id=self.upscale_engine,
+            request_id=request_id,
+            prompt=prompts,
+            image=image_parameters
+        )
 
-        return self.emit_request(prompt=prompts, image_parameters=image_parameters, engine_id=self.upscale_engine)
+        return rq
     
+    def _make_enhance_request(
+        self,
+        weight: int = 0.7,
+        engine_id: str = None,
+        request_id: str = None,
+    ) -> generation.Request:
+        """
+        Create an enhance request for a chain of engines.
+
+        Optional parameters for enhance method:
+
+        :param weight: Weight of the enhancement. 0 is no enhancement, 1 is full enhancement.
+
+        :return: generation.Request object.
+        """
+        extras = Struct()
+        extras.update({
+            'weight': weight
+        })
+
+        image_parameters=generation.ImageParameters()
+        
+        if not request_id:
+            request_id = str(uuid.uuid4())
+        if not engine_id:
+            engine_id = self.enhance_engine
+
+        rq = generation.Request(
+            engine_id=engine_id,
+            request_id=request_id,
+            image=image_parameters,
+            extras=extras
+        )
+        return rq    
 
     # The motivation here is to facilitate constructing requests by passing protobuf objects directly.
     def emit_request(
```

Comment:
do we still need this method? i think we should push up the rq construction into a `_make_generation_request` function for consistency and then pass that to run_request. would be great if we had more info on how people were using the sdk, but i don't think it's worth instrumenting it with telemetry. do we demonstrate using `emit_request` in documentation or tutorials? if it's likely something only used internally by the SDK, we can probably combine it with run_request pretty safely.

Reply:
Yes, I was hesitant to scrap emit_request because we don't know how people are calling the sdk.
I'm not seeing emit_request in any of our tutorials or notebooks, so your suggestion is much more consistent and I will implement it.

Reply:
I've kept the emit_request method  (now unused) and added a deprecation warning.

Maybe we can remove all the deprecated features once we release v1.0.0 😊